### PR TITLE
[sqllab] allow gamma users to access their SQLLab views

### LINKED ISSUE #2798: [permissions] when a user creates a table from sql-lab, make them an owner
currently, the user doesn't get added as an owner after they create a datasource via the visualize flow in sql lab. 

create-datasource-from-sql-lab-flow

expected behavior is that the user be added as an owner on that new table.

### PR DESCRIPTION
**Pull request by mistercrunch**
closes https://github.com/apache/incubator-superset/issues/2798

### CODE REVIEW

**Comment by michellethomas** [comment]
We've seen people run into this issue often, thanks for addressing this issue! cc: @john-bodley

File: superset/security.py [hunk 1]
```diff
@@ -130,8 +131,13 @@ def datasource_access_by_name(
 
         datasources = ConnectorRegistry.query_datasources_by_name(
             db.session, database, datasource_name, schema=schema)
+        print(datasources)
```
Comment (john-bodley) [inline superset/security.py]: Nit. Remove print statements.

**Review by john-bodley — COMMENTED** [review]
Thanks for fixing this. It seems like your PR also cleans ups some other issues around non-unique table names etc.

I agree with the logic that this should reside in the security model, the only potential caveat is we’re adding a new access pattern based on ownership, and thus just wanted to check there was no corner cases which could violate access controls, ie one could potentially add users to the list of owners to bypass the existing security protocol.

File: superset/security.py (hunk 1 above)
Comment (mistercrunch) [reply superset/security.py]: Ooops, I need to add a commit hook that checks for these.

**Comment by john-bodley** [comment]
Sorry I think this PR only remedies part of the problem. Fundamentally the `DashboardFilter`, `SliceFilter`, and `DatasourceFilter` use the corresponding `perm` field to filter out ineligible entities. The `perm` filter is datasource (and not schema etc. specific) and thus a Gamma user can only see datasources which have been explicitly defined for the role.

This PR remedies the problem somewhat by also including datasources owned by the current user (which may not match the permission filter), but does not address the issue of other Gamma users who may be entitled to access it. I wonder if there's a solution which correctly addresses this issue, i.e., the filters correctly list the entity the current user is able to access.

**Comment by ganesh8614** [comment]
Hi @mistercrunch 

I also faced similar issue for gamma user and while trying to modify the files i don't see them.
**Files**
  1) superset/assets/spec/javascripts/sqllab/VisualizeModal_spec.jsx 
2) superset/assets/src/SqlLab/actions.js
3)  superset/assets/src/SqlLab/components/VisualizeModal.jsx 

i have installed and verified in both 0.28v & 0.34v (which is latest)

**Comment by umesh11111** [comment]
you can use the below code and place this code under the views/core.py file. and Gamma user will be able to see his chart .

Class name is slicefilter

def apply(self, query, func): # noqa
if security_manager.all_datasource_access():
return query
perms = self.get_view_menus("datasource_access")
Dash = models.Dashboard
User = ab_models.User
Slice = models.Slice # noqa

    # TODO(bogdan): add `schema_access` support here
    user_roles = [role.name.lower() for role in list(self.get_user_roles())]
    datasource_perms = self.get_view_menus("datasource_access")
    all_datasource_access = security_manager.all_datasource_access()
    owner_ids_query = (
        db.session.query(Slice.id)
        .join(Slice.owners)
        .filter(User.id == User.get_user_id())
    )

    query = query.filter(
        or_(
            Slice.id.in_(owner_ids_query),
        )
    )

    return query
Let me know if you are facing any problem..

**Pull request closed without merging**